### PR TITLE
[CLI] Docker Support

### PR DESCRIPTION
**Pull request by maximsmol**
## Why are these changes needed?

Currently the CLI experience is only optimized on the AWS + SSH backend, this PR adds support for Docker command runners.

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/_private/updater.py [hunk 1]
```diff
@@ -314,7 +316,7 @@ def do_update(self):
             cli_logger.print(
                 "Configuration already up to date, "
                 "skipping file mounts, initalization and setup commands.",
-                _numbered=("[]", "2-5", 6))
+                _numbered=("[]", "2-6", 7))
```
Comment (richardliaw) [inline python/ray/autoscaler/_private/updater.py:319]: NUM_SETUP_STEPS?

**Review by richardliaw — COMMENTED** [review]
changes look ok to me, @ijrsvt please take a look?

File: python/ray/tests/test_cli_patterns/test_ray_up_docker_config.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+auth:
+    ssh_user: ubuntu
+cluster_name: test-cli
+docker:
+    image: rayproject/ray:0.8.7
+    container_name: raydocker
+    pull_before_run: True
```
Comment (ijrsvt) [inline python/ray/tests/test_cli_patterns/test_ray_up_docker_config.yaml:7]: Could you use ray:1.0.0?

**Review by ijrsvt — APPROVED** [review]
Looks good overall! 
Thanks for doing this!

**Review by richardliaw — APPROVED** [review]

**Pull request merged**